spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/jms/activemq/ActiveMQAutoConfigurationTests.java

```diff
@@ -62,6 +62,17 @@ public void configurationBacksOffWhenCustomConnectionFactoryExists() {
 	}
 
 	@Test
+	public void customActiveMQConnectionFactoryConfiguration() {
+		load(EmptyConfiguration.class,
+				"spring.activemq.properties.nonBlockingRedelivery:true",
+				"spring.activemq.properties.producer-window-size:10");
+		ActiveMQConnectionFactory connectionFactory = this.context
```

Comment:
... and to be clear, that would be enumerating all of the properties from `ActiveMQConnectionFactory`:

- `disableTimeStampsByDefault`
- `optimizedMessageDispatch`
- `optimizeAcknowledgeTimeOut`
- ... (and some 30+ others)...

on `ActiveMQProperties` (directly)?